zfsiostats.py and cache

If I have cache devices in my ZFS storage pool I get errors:
2015-10-28 21:20:34,887 tcollector[14141] WARNING: zfsiostats.py:   File ".../tcollector/collectors/0/zfsiostats.py", line 167, in main
2015-10-28 21:20:34,887 tcollector[14141] WARNING: zfsiostats.py:     "expecting last state T_SEPARATOR, now got %s" % ltype
2015-10-28 21:20:34,887 tcollector[14141] WARNING: zfsiostats.py: AssertionError: expecting last state T_SEPARATOR, now got 5

Looks like it doesn't expect to find the cache section

My workaround is adding these lines:
        elif line.startswith("cache"):
            assert ltype in (T_POOL, T_DEVICE), \
                "expecting last state T_POOL or T_DEVICE, now got %s" % ltype
            ltype = T_LEG

--- a/collectors/0/zfsiostats.py
+++ b/collectors/0/zfsiostats.py
@@ -83,7 +83,8 @@ def extract_info(line):
     # 1k blocks
     s_df["used"] = convert_to_bytes(alloc) / 1024
     s_df["available"] = convert_to_bytes(free) / 1024
-    s_df["total"] = s_df["used"] + s_df["available"]
+    # If required, total should be done by front-end tool, as grabbed values are more or less rounded by zpool iostat
+    #s_df["total"] = s_df["used"] + s_df["available"]
 
     s_io = {}
     # magnitudeless variable
@@ -179,8 +180,10 @@ def main():
             ltype = T_DEVICE
         else:
             # must be a pool name
-            assert ltype == T_SEPARATOR, \
-                "expecting last state T_SEPARATOR, now got %s" % ltype
+            #assert ltype == T_SEPARATOR, \
+            #    "expecting last state T_SEPARATOR, now got %s" % ltype
+            if ltype == T_SEPARATOR:
+                parentpoolname = ""
             ltype = T_POOL
 
         if ltype == T_START:
@@ -194,6 +197,10 @@ def main():
         elif ltype == T_POOL:
             line = line.strip()
             poolname, s_df, s_io = extract_info(line)
+            if parentpoolname == "":
+                parentpoolname = poolname
+            else:
+                poolname=parentpoolname+"."+poolname
             capacity_stats_pool[poolname] = s_df
             io_stats_pool[poolname] = s_io
             # marker for leg

